public/app/features/dashboard/api/v2.ts

```diff
@@ -128,12 +127,16 @@ export class K8sDashboardV2API
     if (obj.metadata.name) {
       // remove resource version when updating
       delete obj.metadata.resourceVersion;
-      return this.client.update(obj).then((v) => this.asSaveDashboardResponseDTO(v));
+      return this.client
+        .update(obj)
+        .then((v) => this.asSaveDashboardResponseDTO(v as DashboardWithAccessInfo<DashboardV2Spec>));
     }
-    return await this.client.create(obj).then((v) => this.asSaveDashboardResponseDTO(v));
+    return await this.client
+      .create(obj)
+      .then((v) => this.asSaveDashboardResponseDTO(v as DashboardWithAccessInfo<DashboardV2Spec>));
```

Comment:
🗒️  Not for this PR, but we are adding technical debt here, in the future we should create a dashboard resource client to avoid  using the `as` operator